src/App/component/CodeBasePage.jsx

```diff
@@ -76,39 +97,65 @@ function CodeBasePage(prop) {
     }
   }
 
-  const getCommitFromGitlab = () => {
+  const getCommitFromGitLab = () => {
     const gitlabRepo = currentProject.repositoryDTOList.find(repo => repo.type === 'gitlab')
     if(gitlabRepo !== undefined){
       const query = gitlabRepo.url.split("gitlab.com/")[1]
       Axios.post(`http://localhost:9100/pvs-api/gitlab/commits/${query}`, "",
       {headers: {"Authorization": `${jwtToken}`}})
       .then(() => {
-          Axios.get(`http://localhost:9100/pvs-api/gitlab/commits/${query}`,
-          {headers: {"Authorization": `${jwtToken}`}})
-          .then((response) => {
-              setCommitListData(prevArray => ([...prevArray, ...response.data]))
-          })
-          .catch((e) => {
-              alert(e)
-              console.error(e)
-          })
+        getGitLabCommitFromDB()
+        setLoading(false)
       })
       .catch((e) => {
-          alert(e.response.status)
-          console.error(e)
+        alert(e.response.status)
+        console.error(e)
+      })
+    }
+  }
+
+  const getGitLabCommitFromDB = () => {
+    const gitlabRepo = currentProject.repositoryDTOList.find(repo => repo.type === 'gitlab')
+    if(gitlabRepo !== undefined){
+      const query = gitlabRepo.url.split("gitlab.com/")[1]
+      Axios.get(`http://localhost:9100/pvs-api/gitlab/commits/${query}`,
+      {headers: {"Authorization": `${jwtToken}`}})
+      .then((response) => {
+        setCommitListData(previousArray =>[...previousArray, ...response.data])
```

Comment:
Suggest check if `data` is exist in `response` and `response.data` is a iterable object before using `...`.
```suggestion
        if (response?.data) {
          setCommitListData(previousArray => [...previousArray, ...response.data])
        }
```